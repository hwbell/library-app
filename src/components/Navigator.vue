<template>
  <div id="navigator">
    
    <p class="app-name">Your Library</p>

    <div class="links-holder">
      <p class="nav-link">
        <a href="#reading-list" v-smooth-scroll>my list</a>
      </p>
      <p class="nav-link">
        <a href="#search" v-smooth-scroll>find more</a>
      </p>
    </div>
    
  </div>
</template>

<script>
export default {
  name: "navigator",
  props: {
    //
  },
  data() {
    return {
      //
    };
  },
  methods: {
    //
  }
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style lang="scss" scoped>
#navigator {
  
  z-index: 1;
  background-color: white;
  position: fixed;
  top: 0px;
  left: 0px;
  width: 100%;
  display: flex;
  padding: 0px 40px;
  justify-content: space-between;
  align-items: center;
}
.app-name {
  font-size: 24px;
  margin-left: 25px;
  // padding: 0px;
}
.links-holder {
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
}
.nav-link > a {
  cursor: pointer;
  margin: 15px;
  font-size: 24px;
  &:hover {
    text-decoration: none;
  }
}
</style>