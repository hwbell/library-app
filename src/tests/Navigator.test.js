// Import Vue and the component being tested
import { shallowMount, mount } from '@vue/test-utils'

import Vue from 'vue'
import Navigator from '../components/Navigator.vue';

import vueSmoothScroll from "vue2-smooth-scroll";
Vue.use(vueSmoothScroll);

describe('Navigator', () => {

  it('renders correctly', () => {
    const wrapper = mount(Navigator)
    expect(wrapper.element).toMatchSnapshot()
  })

  const wrapper = shallowMount(Navigator)
  it('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  it('contains navigator selector', () => {
    expect(wrapper.html()).toContain('id="navigator"')
  })
  it('contains links-holder selector', () => {
    expect(wrapper.html()).toContain('class="links-holder"')
  })
  

})

