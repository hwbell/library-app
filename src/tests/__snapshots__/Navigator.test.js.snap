// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigator renders correctly 1`] = `
<div
  id="navigator"
>
  <p
    class="app-name"
  >
    Your Library
  </p>
   
  <mqshow
    if="sm"
  >
    <b-dropdown
      class="m-md-2 dropdown"
      id="dropdown"
      text="Go to"
      variant="primary"
    >
      <b-dropdown-item>
        <a
          href="#reading-list"
        >
          my list
        </a>
      </b-dropdown-item>
       
      <b-dropdown-item>
        <a
          href="#search"
        >
          find more
        </a>
      </b-dropdown-item>
    </b-dropdown>
  </mqshow>
   
  <mqshow
    if="md!"
  >
    <div
      class="links-holder"
    >
      <p
        class="nav-link"
      >
        <a
          href="#reading-list"
        >
          my list
        </a>
      </p>
       
      <p
        class="nav-link"
      >
        <a
          href="#search"
        >
          find more
        </a>
      </p>
    </div>
  </mqshow>
</div>
`;
